modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -261,8 +263,9 @@ public <R> Map<ClusterNode, JobExecution<R>> broadcastAsync(
         return map;
     }
 
-    private CompletableFuture<PayloadInputChannel> executeOnOneNode(
-            ClusterNode node,
+    private CompletableFuture<PayloadInputChannel> executeOnNodesAsync(
+            ClusterNode coordinatorNode,
```

Comment:
`Coordinator` does not make sense.
* For `execute` it is not needed, we can use any node from the set (and we should pick one of the nodes where a direct client connection exists, if possible)
* For `broadcast` we only have one node in the set anyway

Reply:
The best approach would be to pass only a collection of nodes to this method, then pick `preferredNode` for the `serviceAsync` according to the existing connections, or `null` if there is no intersection of passed nodes and connections.

Reply:
> Do you think it's worth randomizing the selection
Yes, for load balancing.